sklearn/utils/estimator_checks.py

```diff
@@ -397,6 +399,52 @@ def check_dtype_object(name, Estimator):
 
 
 @ignore_warnings
+def check_dict_unchanged(name, Estimator):
+    # these two estimators change the state of __dict__
+    # and need to be fixed
+    if name in ['NMF', 'ProjectedGradientNMF']:
```

Comment:
Actually, it might be simpler to remove [this line](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/decomposition/nmf.py#L1109) instead, that should get rid of the issue.